src/primitives/sinsemilla.rs

```diff
@@ -20,6 +21,15 @@ fn lebs2ip_k(bits: &[bool]) -> u32 {
         .fold(0u32, |acc, (i, b)| acc + if *b { 1 << i } else { 0 })
 }
 
+/// The sequence of K bits in little-endian order representing an integer
+/// up to `2^K` - 1.
+pub fn i2lebsp_k(int: usize) -> [bool; K] {
+    assert!(int < (1 << K));
+    gen_const_array(int, |int: &mut usize, mask: usize| {
+        ((*int & (1 << mask)) >> mask) == 1
```

Comment:
This looks to me like it's the same as `(*int & (1 << mask)) != 0` - is this version more performant?

Reply:
```rs
fn bar(a: usize, mask: usize) -> bool {
    ((a & (1 << mask)) >> mask) == 1
}

fn foo(a: usize, mask: usize) -> bool {
    (a & (1 << mask)) != 0
}
```
```
playground::bar:
	movq	%rsi, %rcx
	movl	$1, %eax
	shlq	%cl, %rax
	andq	%rdi, %rax
	shrq	%cl, %rax
	cmpq	$1, %rax
	sete	%al
	retq

playground::foo:
	btq	%rsi, %rdi
	setb	%al
	retq
```
It looks like the `!= 0` test enables LLVM to figure out this is bit-testing, and swap in that opcode. So that is the faster operation to use.

Reply:
Made this change alongside my suggestions.